mne/io/brainvision/brainvision.py

```diff
@@ -79,30 +80,49 @@ def __init__(self, vhdr_fname, montage=None,
                  event_id=None, verbose=None):  # noqa: D102
         # Channel info and events
         logger.info('Extracting parameters from %s...' % vhdr_fname)
-        vhdr_fname = os.path.abspath(vhdr_fname)
         info, fmt, self._order, mrk_fname, montage = _get_vhdr_info(
             vhdr_fname, eog, misc, scale, montage)
         events = _read_vmrk_events(mrk_fname, event_id, response_trig_shift)
         _check_update_montage(info, montage)
         with open(info['filename'], 'rb') as f:
-            f.seek(0, os.SEEK_END)
-            n_samples = f.tell()
-        dtype_bytes = _fmt_byte_dict[fmt]
+            if isinstance(fmt, dict):  # ASCII, this will be slow :(
+                n_skip = 0
+                for ii in range(int(fmt['skiplines'])):
+                    n_skip += len(f.readline())
+                offsets = np.cumsum([n_skip] + [len(line) for line in f])
+                n_samples = len(offsets) - 1
+            else:
+                f.seek(0, os.SEEK_END)
+                n_samples = f.tell()
+                dtype_bytes = _fmt_byte_dict[fmt]
+                offsets = None
+                n_samples = n_samples // (dtype_bytes * (info['nchan'] - 1))
         self.preload = False  # so the event-setting works
-        last_samps = [(n_samples // (dtype_bytes * (info['nchan'] - 1))) - 1]
-        self._create_event_ch(events, last_samps[0] + 1)
+        self._create_event_ch(events, n_samples)
         super(RawBrainVision, self).__init__(
-            info, last_samps=last_samps, filenames=[info['filename']],
-            orig_format=fmt, preload=preload, verbose=verbose)
+            info, last_samps=[n_samples - 1], filenames=[info['filename']],
+            orig_format=fmt, preload=preload, verbose=verbose,
+            raw_extras=[offsets])
 
     def _read_segment_file(self, data, idx, fi, start, stop, cals, mult):
         """Read a chunk of raw data."""
         # read data
-        dtype = _fmt_dtype_dict[self.orig_format]
-        n_data_ch = len(self.ch_names) - 1
-        _read_segments_file(self, data, idx, fi, start, stop, cals, mult,
-                            dtype=dtype, n_channels=n_data_ch,
-                            trigger_ch=self._event_ch)
+        if isinstance(self.orig_format, string_types):
+            dtype = _fmt_dtype_dict[self.orig_format]
+            n_data_ch = len(self.ch_names) - 1
+            _read_segments_file(self, data, idx, fi, start, stop, cals, mult,
+                                dtype=dtype, n_channels=n_data_ch,
+                                trigger_ch=self._event_ch)
+        else:
+            offsets = self._raw_extras[fi]
+            with open(self._filenames[fi], 'rb') as fid:
+                fid.seek(offsets[start])
+                block = np.empty((len(self.ch_names), stop - start))
+                for ii in range(stop - start):
+                    line = fid.readline().strip().split().replace(',', '.')
```

Comment:
This should probably be ``line = fid.readline().strip().replace(',', '.').split()``